test/parallel/test-stream-readable-to-web.mjs

```diff
@@ -0,0 +1,64 @@
+import { mustCall } from '../common/index.mjs';
+import { Readable } from 'node:stream';
+import { memoryUsage } from 'node:process';
+import assert from 'node:assert';
+import { setImmediate } from 'node:timers/promises';
+
+// Based on: https://github.com/nodejs/node/issues/46347#issuecomment-1413886707
+// edit: make it cross-platform as /dev/urandom is not available on Windows
+
+const MAX_MEM = 256 * 1024 * 1024; // 256 MiB
+
+function checkMemoryUsage() {
+  assert(memoryUsage().arrayBuffers < MAX_MEM);
+}
+
+const MAX_BUFFERS = 1000;
+let buffersCreated = 0;
+
+const randomNodeStream = new Readable({
+  read(size) {
+    if (buffersCreated >= MAX_BUFFERS) {
+      this.push(null);
+      return;
+    }
+
+    this.push(Buffer.alloc(size));
+    buffersCreated++;
+  }
+});
+
+randomNodeStream.on('error', (err) => {
+  assert.fail(err);
+});
+
+// Before doing anything, make sure memory usage is okay
+checkMemoryUsage();
+
+// Create stream and check memory usage remains okay
+
+const randomWebStream = Readable.toWeb(randomNodeStream);
+
+checkMemoryUsage();
+
+let timeout;
+try {
+  // Wait two seconds before consuming the stream to see if memory usage increases
+  timeout = setTimeout(mustCall(async () => {
+    // Did the stream leak memory?
+    checkMemoryUsage();
+    // eslint-disable-next-line no-unused-vars
+    for await (const _ of randomWebStream) {
+      // Yield event loop to allow garbage collection
```

Comment:
FWIW, this behavior likely relies on the bug described in https://github.com/nodejs/node/issues/54918 - the event loop is not supposed to block on just any garbage collection, which can cause a deadlock, and in the future it might go away (if someone manages to eradicate the deadlock by rewriting the task runner).

Alternatively if there's a specific leaky object with a known class that this tries to test, `checkIfCollectableByCounting()` from `test/common/gc.js` might be less flaky. Though I can't really tell whether that's applicable from a look of this test.

Reply:
Good to know. I agree that issue seems related here; particularly with the previous heavy use of timers. I believe that the Readable / GC is doing what they are supposed to regarding memory management.